packages/safe-tools-client/app/templates/application.hbs

```diff
@@ -46,7 +46,12 @@
 </Boxel::Dashboard>
 
 <NetworkConnectModal
-  @isOpen={{and this.isShowingConnectModal (not this.wallet.isConnected)}}
+  @isOpen={{and (and this.isShowingConnectModal (not this.wallet.isConnected)) (not this.wallet.isConnectUnsupportedNetwork)}}
   @onClose={{set this 'isShowingConnectModal' false}}
   @isConnected={{this.wallet.isConnected}}
-/>
\ No newline at end of file
+/>
+
+<UnsupportedNetworkModal
+  @isOpen={{this.wallet.isConnectUnsupportedNetwork}}
+  @onClose={{set this.wallet 'isConnectUnsupportedNetwork' false}}
```

Comment:
Is it conceptually correct to set `isConnectUnsupportedNetwork` to false even when the user is in fact perhaps still connected to an unsupported network?